### ISSUE TITLE
Benchmark list implementations in collect classes

### ISSUE THREAD

**Issue by heuermh** [heuermh/dishevelled-bio#33]

**Comment by heuermh** [comment]
Results on OSX
```
CEUTrio.HiSeq.WGS.b37.NA12878.20.21.10k.sam 5.3M

Benchmark                                               Mode  Cnt   Score   Error  Units
SamCollectBenchmarks.collectSamAppendOnlyLinkedList    thrpt   25  15.429 ± 0.266  ops/s
SamCollectBenchmarks.collectSamLinkedList              thrpt   25  15.243 ± 0.325  ops/s
SamCollectBenchmarks.collectSamSmallCapacityArrayList  thrpt   25  15.142 ± 0.152  ops/s
SamCollectBenchmarks.collectSamLargeCapacityArrayList  thrpt   25  11.935 ± 0.118  ops/s

CEUTrio.HiSeq.WGS.b37.NA12878.20.21.sam 354M

java -jar target/dsh-bio-benchmarks.jar org.dishevelled.bio.benchmarks.SamCollectBenchmarks*

Benchmark                                               Mode  Cnt  Score   Error  Units
SamCollectBenchmarks.collectSamSmallCapacityArrayList  thrpt   25  0.200 ± 0.002  ops/s
SamCollectBenchmarks.collectSamLargeCapacityArrayList  thrpt   25  0.199 ± 0.005  ops/s
SamCollectBenchmarks.collectSamAppendOnlyLinkedList    thrpt   25  0.187 ± 0.002  ops/s
SamCollectBenchmarks.collectSamLinkedList              thrpt   25  0.185 ± 0.003  ops/s

java -Xms14g -Xmx16g -jar target/dsh-bio-benchmarks.jar org.dishevelled.bio.benchmarks.SamCollectBenchmarks*

Benchmark                                               Mode  Cnt  Score   Error  Units
SamCollectBenchmarks.collectSamLargeCapacityArrayList  thrpt   25  0.217 ± 0.004  ops/s
SamCollectBenchmarks.collectSamSmallCapacityArrayList  thrpt   25  0.201 ± 0.003  ops/s
SamCollectBenchmarks.collectSamAppendOnlyLinkedList    thrpt   25  0.193 ± 0.004  ops/s
SamCollectBenchmarks.collectSamLinkedList              thrpt   25  0.189 ± 0.002  ops/s

java -Xms24g -Xmx32g -jar target/dsh-bio-benchmarks.jar org.dishevelled.bio.benchmarks.SamCollectBenchmarks*

Benchmark                                               Mode  Cnt  Score   Error  Units
SamCollectBenchmarks.collectSamLargeCapacityArrayList  thrpt   25  0.208 ± 0.005  ops/s
SamCollectBenchmarks.collectSamSmallCapacityArrayList  thrpt   25  0.196 ± 0.002  ops/s
SamCollectBenchmarks.collectSamAppendOnlyLinkedList    thrpt   25  0.191 ± 0.005  ops/s
SamCollectBenchmarks.collectSamLinkedList              thrpt   25  0.185 ± 0.003  ops/s
```

**Comment by heuermh** [comment]
On linux
```
CEUTrio.HiSeq.WGS.b37.NA12878.20.21.10k.sam 5.3M

Benchmark                                               Mode  Cnt   Score   Error  Units
SamCollectBenchmarks.collectSamAppendOnlyLinkedList    thrpt   25  14.138 ± 0.259  ops/s
SamCollectBenchmarks.collectSamSmallCapacityArrayList  thrpt   25  13.894 ± 0.092  ops/s
SamCollectBenchmarks.collectSamLinkedList              thrpt   25  13.730 ± 0.195  ops/s
SamCollectBenchmarks.collectSamLargeCapacityArrayList  thrpt   25  13.445 ± 0.131  ops/s

CEUTrio.HiSeq.WGS.b37.NA12878.20.21.sam 354M

java -jar target/dsh-bio-benchmarks.jar org.dishevelled.bio.benchmarks.SamCollectBenchmarks*

Benchmark                                               Mode  Cnt  Score   Error  Units
SamCollectBenchmarks.collectSamLinkedList              thrpt   25  0.100 ± 0.016  ops/s
SamCollectBenchmarks.collectSamSmallCapacityArrayList  thrpt   25  0.095 ± 0.016  ops/s
SamCollectBenchmarks.collectSamLargeCapacityArrayList  thrpt   25  0.094 ± 0.015  ops/s
SamCollectBenchmarks.collectSamAppendOnlyLinkedList    thrpt   25  0.084 ± 0.012  ops/s

java -Xms14g -Xmx16g -jar target/dsh-bio-benchmarks.jar org.dishevelled.bio.benchmarks.SamCollectBenchmarks*

Benchmark                                               Mode  Cnt  Score   Error  Units
SamCollectBenchmarks.collectSamSmallCapacityArrayList  thrpt   25  0.141 ± 0.012  ops/s
SamCollectBenchmarks.collectSamLinkedList              thrpt   25  0.139 ± 0.008  ops/s
SamCollectBenchmarks.collectSamAppendOnlyLinkedList    thrpt   25  0.132 ± 0.008  ops/s
SamCollectBenchmarks.collectSamLargeCapacityArrayList  thrpt   25  0.118 ± 0.007  ops/s

java -Xms24g -Xmx32g -jar target/dsh-bio-benchmarks.jar org.dishevelled.bio.benchmarks.SamCollectBenchmarks*

Benchmark                                               Mode  Cnt  Score   Error  Units
SamCollectBenchmarks.collectSamLinkedList              thrpt   25  0.152 ± 0.011  ops/s
SamCollectBenchmarks.collectSamSmallCapacityArrayList  thrpt   25  0.147 ± 0.017  ops/s
SamCollectBenchmarks.collectSamLargeCapacityArrayList  thrpt   25  0.136 ± 0.020  ops/s
SamCollectBenchmarks.collectSamAppendOnlyLinkedList    thrpt   25  0.132 ± 0.018  ops/s
```

**Comment by heuermh** [comment]
Added benchmark in commit cdee204c94341db7489f8f42c76d5693bbfc5984, fixed in commit 33a4ae98a0f41ba4dd05eb9a9c063762bddf1f9b
